management/src/main/scala/akka/management/HealthCheckRoutes.scala

```diff
@@ -40,6 +40,22 @@ private[akka] class HealthCheckRoutes(system: ExtendedActorSystem) extends Manag
   }
 
   override def routes(mrps: ManagementRouteProviderSettings): Route = {
+    if (settings.enabled) {
+      concat(
+          path(PathMatchers.separateOnSlashes(settings.readinessPath)) {
+            get {
+              onComplete(healthChecks.ready())(healthCheckResponse)
+            }
+          },
+          path(PathMatchers.separateOnSlashes(settings.livenessPath)) {
+            get {
+              onComplete(healthChecks.alive())(healthCheckResponse)
+            }
+          })
+    } else complete(StatusCodes.OK)
```

Comment:
Not sure if the right thing to do. I was looking for a Route that could simply make this compile but with zero effect. 

I guess that if we add this here, all subsequential routes will be ignored. Can someone confirm it? What are the alternatives here?

One possible solution is to add the `akka.management.HealthCheckRoutes` programmatically if the flag is enabled (instead of using `route-providers`).

Reply:
yes, this is probably not correct

@jrudolph is there a noop (non-matching) `Route`?

Reply:
It's `reject` so the request backtracks out of that branch and tries the next ones.

Reply:
thanks